src/lib/utils.ts

```diff
@@ -87,32 +84,102 @@ export const promiseErrorMultiplier = <T, Q extends Error>(
   });
 };
 
+export const generateGatewayMap = async (
+  l2Multicaller: MultiCaller,
+  l2Network: L2Network,
+  l1Provider: Provider
+) => {
+  const fromBlock = 0;
```

Comment:
here we can start at the L1 block number of the L2 genesis block instead of zero (might be nice to add this field to the networks object in arbitrum-sdk in fact 🤔 )

Reply:
That's a good idea, but default gateway or others gateway may be set before genesis?

Reply:
ah good point; let's just leave as is for now then